cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/consumption/AwsS3ConsumptionCalculator.java

```diff
@@ -0,0 +1,105 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.consumption;
+
+import java.util.Comparator;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.validation.ValidationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.amazonaws.services.cloudwatch.model.Datapoint;
+import com.amazonaws.services.cloudwatch.model.GetMetricStatisticsResult;
+import com.cloudera.thunderhead.service.metering.events.MeteringEventsProto;
+import com.sequenceiq.cloudbreak.cloud.ConsumptionCalculator;
+import com.sequenceiq.cloudbreak.cloud.aws.common.connector.resource.AwsCloudWatchCommonService;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.model.CloudConsumption;
+import com.sequenceiq.cloudbreak.cloud.model.objectstorage.ObjectStorageSizeRequest;
+import com.sequenceiq.cloudbreak.cloud.model.objectstorage.ObjectStorageSizeResponse;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.cloudbreak.common.mappable.StorageType;
+import com.sequenceiq.common.model.FileSystemType;
+
+@Service
+public class AwsS3ConsumptionCalculator implements ConsumptionCalculator {
+
+    private static final StorageType S3 = StorageType.S3;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsS3ConsumptionCalculator.class);
+
+    private static final int DATE_RANGE_WIDTH_IN_DAYS = 2;
+
+    private static final long NO_BYTE_IN_MB = 1000L * 1000L;
+
+    @Inject
+    private AwsCloudWatchCommonService cloudWatchCommonService;
+
+    @Override
+    public void validate(CloudConsumption cloudConsumption) throws ValidationException {
+        String storageLocation = cloudConsumption.getStorageLocation();
+        if (storageLocation == null || !storageLocation.startsWith(FileSystemType.S3.getProtocol())) {
+            throw new ValidationException(String.format("Storage location must start with '%s' if required file system type is '%s'!",
+                    FileSystemType.S3.getProtocol(), FileSystemType.S3.name()));
+        }
+    }
+
+    @Override
+    public ObjectStorageSizeResponse calculate(ObjectStorageSizeRequest request) {
```

Comment:
StorageSizeRequest and StorageSizeResponse ?

Reply:
yes I forget)corrected